build/builder.go

```diff
@@ -693,3 +728,87 @@ func (b *Builder) writeShard(fn string, ib *zoekt.IndexBuilder) (*finishedShard,
 
 // umask holds the Umask of the current process
 var umask os.FileMode
+
+// MergeMeta updates the .meta files for the shards on disk for o.
+//
+// Note: currently we only merge the RawConfig field. If a raw config field
+// exists on disk but not in the options, we keep the on disk value.
+func MergeMeta(o *Options) error {
+	// TODO should this logic live in the zoekt pkg rather than the build pkg?
+	// Argument for build is its the only place we deal with writing to multiple
+	// shards. Argument for zoekt is it's the only place that understands meta
+	// implementation.
+	var todo map[string]string
+	for i := 0; ; i++ {
+		fn := o.shardName(i)
```

Comment:
My bad internet lost a few comments :( I wrote agood response here, but will be more brief now. builder relies on shardName(int) since it doesn't know how many shards it will create. Additionally there may be no shards on disk (new repo). So shardNames() helper will only be used in this bit of code here. To me it seems clearer to just keep using the current signature.

However, I'll probably move this function out based on your suggestion so will try make this cleaner.